org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed

Exception in plugin Quarkus Tools (1.11.0.229)
IntelliJ IDEA 2022.1.1 Build #IC-221.5591.52, built on May 10, 2022

org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:122)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.IOException: Stream closed
	at java.base/java.io.BufferedInputStream.getBufIfOpen(BufferedInputStream.java:176)
	at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:275)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:79)
	... 6 more
	
	
Following exceptions happened soon after this one, most probably they are induced.


2022-05-31T12:45:05.779
com.intellij.util.IncorrectOperationException: Sorry but parent: org.jetbrains.idea.maven.MavenDisposable@7d9fc353 (class org.jetbrains.idea.maven.MavenDisposable) has already been disposed (see the cause for stacktrace) so the child: org.jetbrains.idea.maven.server.MavenServerManager$$Lambda$3246/0x0000000801fee840@3e424548 (class org.jetbrains.idea.maven.server.MavenServerManager$$Lambda$3246/0x0000000801fee840) will never be disposed
	at com.intellij.openapi.util.ObjectTree.register(ObjectTree.java:53)
	at com.intellij.openapi.util.Disposer.register(Disposer.java:118)
	at org.jetbrains.idea.maven.server.MavenServerManager.registerDisposable(MavenServerManager.java:208)
	at org.jetbrains.idea.maven.server.MavenServerManager.registerNewConnector(MavenServerManager.java:203)
	at org.jetbrains.idea.maven.server.MavenServerManager.doGetOrCreateConnector(MavenServerManager.java:164)
	at org.jetbrains.idea.maven.server.MavenServerManager.getConnector(MavenServerManager.java:135)
	at org.jetbrains.idea.maven.server.MavenServerManager$4.create(MavenServerManager.java:461)
	at org.jetbrains.idea.maven.server.MavenServerManager$4.create(MavenServerManager.java:443)
	at org.jetbrains.idea.maven.server.RemoteObjectWrapper.getOrCreateWrappee(RemoteObjectWrapper.java:42)
	at org.jetbrains.idea.maven.server.MavenEmbedderWrapper.lambda$resolveTransitively$12(MavenEmbedderWrapper.java:239)
	at org.jetbrains.idea.maven.server.RemoteObjectWrapper.performCancelable(RemoteObjectWrapper.java:90)
	at org.jetbrains.idea.maven.server.MavenEmbedderWrapper.resolveTransitively(MavenEmbedderWrapper.java:239)
	at com.redhat.devtools.intellij.quarkus.maven.MavenToolDelegate.ensureDownloaded(MavenToolDelegate.java:143)
	at com.redhat.devtools.intellij.quarkus.maven.MavenToolDelegate.getDeploymentFiles(MavenToolDelegate.java:103)
	at com.redhat.devtools.intellij.quarkus.maven.MavenToolDelegate.getDeploymentFiles(MavenToolDelegate.java:59)
	at com.redhat.devtools.intellij.quarkus.QuarkusModuleUtil.lambda$ensureQuarkusLibrary$0(QuarkusModuleUtil.java:97)
	at com.intellij.openapi.roots.ModuleRootModificationUtil.lambda$updateModel$9(ModuleRootModificationUtil.java:147)
	at com.intellij.openapi.roots.ModuleRootModificationUtil.modifyModel(ModuleRootModificationUtil.java:155)
	at com.intellij.openapi.roots.ModuleRootModificationUtil.updateModel(ModuleRootModificationUtil.java:146)
	at com.redhat.devtools.intellij.quarkus.QuarkusModuleUtil.ensureQuarkusLibrary(QuarkusModuleUtil.java:89)
	at com.redhat.devtools.intellij.quarkus.QuarkusPostStartupActivity.runActivity(QuarkusPostStartupActivity.java:24)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runStartupActivity(StartupManagerImpl.kt:311)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runActivityAndMeasureDuration(StartupManagerImpl.kt:290)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runActivityAndMeasureDuration$default(StartupManagerImpl.kt:285)
	at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$1$1.run(StartupManagerImpl.kt:238)
	at com.intellij.openapi.project.DumbServiceImpl.runWithWaitForSmartModeDisabled(DumbServiceImpl.java:648)
	at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$1.accept(StartupManagerImpl.kt:237)
	at com.intellij.ide.startup.impl.StartupManagerImpl$runPostStartupActivities$1.accept(StartupManagerImpl.kt:62)
	at com.intellij.openapi.extensions.impl.ExtensionPointImpl.processWithPluginDescriptor(ExtensionPointImpl.java:301)
	at com.intellij.openapi.extensions.ExtensionPointName.processWithPluginDescriptor(ExtensionPointName.java:143)
	at com.intellij.ide.startup.impl.StartupManagerImpl.runPostStartupActivities(StartupManagerImpl.kt:232)
	at com.intellij.ide.startup.impl.StartupManagerImpl.access$runPostStartupActivities(StartupManagerImpl.kt:62)
	at com.intellij.ide.startup.impl.StartupManagerImpl$projectOpened$3$1.run(StartupManagerImpl.kt:180)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:189)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:176)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:365)
	at com.intellij.openapi.progress.util.BackgroundTaskUtil.runUnderDisposeAwareIndicator(BackgroundTaskUtil.java:343)
	at com.intellij.ide.startup.impl.StartupManagerImpl$projectOpened$3.run(StartupManagerImpl.kt:180)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)

This is happening more frequently (specially when I close a class):
 
```
org.eclipse.lsp4j.jsonrpc.JsonRpcException: java.io.IOException: Stream closed
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:122)
	at org.eclipse.lsp4j.jsonrpc.json.ConcurrentMessageProcessor.run(ConcurrentMessageProcessor.java:113)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.io.IOException: Stream closed
	at java.base/java.io.BufferedInputStream.getBufIfOpen(BufferedInputStream.java:176)
	at java.base/java.io.BufferedInputStream.read(BufferedInputStream.java:275)
	at org.eclipse.lsp4j.jsonrpc.json.StreamMessageProducer.listen(StreamMessageProducer.java:79)
	... 6 more
```

--- a/build.gradle
+++ b/build.gradle
@@ -146,7 +146,7 @@ dependencies {
         transitive = false
     }
     compile 'org.eclipse.lsp4mp:org.eclipse.lsp4mp.ls:0.5.0'
-    compile 'org.eclipse.lsp4j:org.eclipse.lsp4j:0.14.0'
+    compile 'org.eclipse.lsp4j:org.eclipse.lsp4j:0.15.0'
     //required by lsp4j as the version from IJ is incompatible
     compile 'com.google.code.gson:gson:2.8.9'
     compile 'com.vladsch.flexmark:flexmark:0.50.50'

--- a/src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LSPIJUtils.java
+++ b/src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LSPIJUtils.java
@@ -51,7 +51,6 @@ public class LSPIJUtils {
         param.setPosition(start);
         TextDocumentIdentifier id = new TextDocumentIdentifier();
         if (uri != null) {
-            param.setUri(uri.toString());
             id.setUri(uri.toString());
         }
         param.setTextDocument(id);
@@ -176,7 +175,6 @@ public class LSPIJUtils {
         Position start = toPosition(offset, document);
         CompletionParams param = new CompletionParams();
         param.setPosition(start);
-        param.setUri(fileUri.toString());
         TextDocumentIdentifier id = new TextDocumentIdentifier();
         id.setUri(fileUri.toString());
         param.setTextDocument(id);

--- a/src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LanguageServerWrapper.java
+++ b/src/main/java/com/redhat/devtools/intellij/quarkus/lsp4ij/LanguageServerWrapper.java
@@ -20,6 +20,7 @@ import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.util.messages.MessageBusConnection;
 import com.redhat.devtools.intellij.quarkus.lsp4ij.server.StreamConnectionProvider;
 import org.eclipse.lsp4j.ClientCapabilities;
+import org.eclipse.lsp4j.ClientInfo;
 import org.eclipse.lsp4j.CodeActionCapabilities;
 import org.eclipse.lsp4j.CodeActionKind;
 import org.eclipse.lsp4j.CodeActionKindCapabilities;
@@ -303,7 +304,7 @@ public class LanguageServerWrapper {
                     .setSynchronization(new SynchronizationCapabilities(Boolean.TRUE, Boolean.TRUE, Boolean.TRUE));
             initParams.setCapabilities(
                     new ClientCapabilities(workspaceClientCapabilities, textDocumentClientCapabilities, lspStreamProvider.getExperimentalFeaturesPOJO()));
-            initParams.setClientName(CLIENT_NAME);
+            initParams.setClientInfo(new ClientInfo(CLIENT_NAME));
 
             initParams.setInitializationOptions(this.lspStreamProvider.getInitializationOptions(rootURI));
             initParams.setTrace(this.lspStreamProvider.getTrace(rootURI));

